### DISCUSSION TITLE
psql: error: could not connect to server: FATAL:  role does not exist

### DISCUSSION THREAD

**Discussion by Fukitsu** [simple-login/app#665] [Self-hosting question]
After creating the Postgres database container with
``
sudo docker run -d \
    --name sl-db \
    -e POSTGRES_PASSWORD=custompass\
    -e POSTGRES_USER=customuser \
    -e POSTGRES_DB=simplelogin \
    -p 127.0.0.1:5432:5432 \
    -v $(pwd)/sl/db:/var/lib/postgresql/data \
    --restart always \
    --network="sl-network" \
    postgres:12.1
``
and then running
``sudo docker exec -it sl-db psql -U customuser simplelogin``

I get this error
``psql: error: could not connect to server: FATAL:  role "customuser" does not exist``

However if I run
``sudo docker exec -it sl-db psql -U myuser simplelogin``
it works. What's the problem here?

**Comment by Fukitsu** [comment]
I got it to somehow work by manually adding the user but now when I try to create my account it says the email isn't valid and the confirmation email doesn't arrive and when I try to resend it I get a screen that says 
"Server error
Looks like we are having some server issues...

We are notified and will look at this issue asap!"

**Reply by nguyenkims** [reply]
Can you try to remove the container and retry? You can remove the container with `docker rm -f sl-db`

**Reply by Fukitsu** [reply]
I did so but the same thing happens when trying to sign up/resend confirmation email

**Reply by nguyenkims** [reply]
Can you paste the docker log for `sl-app` here? It might contain more info.

**Comment by justjeff** [comment]
It's an old question, but I ran into the same issue recently.

If the docker command is straight copy-pasted and run *ever*, then `myuser // mypassword` are locked in at `~/sl/db/` even if the docker container is rebuilt.

The fix is *nearly* a full reset.
`docker rm -f sl-db`
`sudo rm -rf sl/db` 
`mkdir sl/db`
Your docker network should still exist, you can then re-run the docker database command(s) with updated user/pass data.
